gtk/gtk.go

```diff
@@ -8516,6 +8516,62 @@ func (v *TreeModelFilter) ConvertIterToChildIter(filterIter *TreeIter) *TreeIter
 	return t
 }
 
+// ConvertPathToChildPath is a wrapper around gtk_tree_model_filter_convert_path_to_child_path().
+func (v *TreeModelFilter) ConvertPathToChildPath(filterPath *TreePath) *TreePath {
+	path := C.gtk_tree_model_filter_convert_path_to_child_path(v.native(), filterPath.native())
+	if path == nil {
+		return nil
+	}
+
+	p := &TreePath{path}
+	runtime.SetFinalizer(p, (*TreePath).free)
+
+	return p
+}
+
+// Refilter is a wrapper around gtk_tree_model_filter_refilter().
+func (v *TreeModelFilter) Refilter() {
+	C.gtk_tree_model_filter_refilter(v.native())
+}
+
+type TreeModelFilterVisibleFunc func(model *TreeModelFilter, iter *TreeIter, userData interface{}) bool
+
+type treeModelFilterVisibleFuncData struct {
+	fn       TreeModelFilterVisibleFunc
+	userData interface{}
+}
+
+var (
+	treeModelVisibleFilterFuncRegistry = struct {
+		sync.RWMutex
+		next int
+		m    map[int]treeModelFilterVisibleFuncData
+	}{
+		next: 1,
+		m:    make(map[int]treeModelFilterVisibleFuncData),
+	}
+)
+
+// SetVisibleFunc is a wrapper around gtk_tree_model_filter_set_visible_func().
+func (v *TreeModelFilter) SetVisibleFunc(f TreeModelFilterVisibleFunc, userData ...interface{}) error {
+	if len(userData) > 1 {
+		return errors.New("userData len must be 0 or 1")
+	}
+
+	t := treeModelFilterVisibleFuncData{fn: f}
+	if len(userData) > 0 {
+		t.userData = userData[0]
```

Comment:
Is it really correct to only set the first value when we are allowing the programmer to pass a variable number of arguments for `userData`?

Reply:
At line 8557, I check that the caller has passed in at most 1 userData argument. I modeled this code on the implementation of Connect for signals (`connectClosure` in glib/connect.go) If you want this generalized, I think it should be a separate issue and be fixed for all callbacks taking user data. However, it's your code base, so I can change this code to handle multiple userData args if you wish.

Reply:
So my point of view is the following.

Let's say I wouldn't know about this implementation and I am doing the following:

```go
a.SetVisibleFunc(someFunc, userDataOne, userDataTwo)
```

Now I would never be able to access userDataTwo again, so why would I be able to pass it at all. I find this to be confusing. Why not just pass a pointer to the array(variadic arguments)?

Reply:
The documentation for `GtkTreeModelFilterVisibleFunc` only allows one item of user data (a gpointer), so I copied how `Connect` did this for signals (with the test that the userData is of length 0 or 1.) I could have a single parameter and test for `nil` if you think that makes things clearer? I'm learning go by porting a Python gtk app, so I'm happy to be guided on points of style.